### ISSUE TITLE
Default container

### ISSUE THREAD

**Issue by comzeradd** [mozilla/testpilot-containers#356]
It would be nice if I could set a default container for new tabs.

So for instance if I click on an external app it opens without any container at all. It feels like there is an extra default container besides the ones I've set.

**Comment by jonathanKingston** [comment]
We have "No container" in various context menus etc and functionality wise that as you say behaves the same way as a container.

Are you essentially looking to change the name, color and icon of "No container"? Or is the issue more around menus not containing "No container" everywhere?

**Comment by comzeradd** [comment]
Yes, I think that the "No container" should also somehow visualized. From a UX perspective I think it would be more clear that actually this is also a container.

**Comment by ruihildt** [comment]
On the other hand, I would like to define myself what is the default container: when I click on "+" to open a tab, I want it to open by default in the container of my choice.

The "No container" is in fact a container, the "Default" one from the user perspective.

**Comment by jimdigriz** [comment]
Most intuative would be 'New Tab' should open a tab in same container as the current selection.  Also, I think pushing home the message that 'No container' is 'personal' will fix a lot of the UI problems here too.

For example I put everything into my personal container, and now [SnoozeTabs](https://github.com/bwinton/SnoozeTabs#readme) and Pocket are un-usable.  By putting them in 'No container' I avoid this problem, but the instructions encourage a 'personal' container when in practice it just does not work out at all well for me.

**Comment by smichel17** [comment]
An alternative resolution would be to make new tabs private by default (#419, #427), potentially with the option to open new tabs in the current container, by default (#245, #342).

**Comment by albertoefg** [comment]
I would like an option to have a default container as soon as Firefox opens, so I won't forget I have containers and I can use them. 😄

**Comment by chicagobuss** [comment]
Hi, just adding my voice here - I use firefox MOSTLY for work but I have a few cases where I need / want to use containers.  So for me, I just want the work container to be the default container - the big thing I'm hoping for is ctrl-t ALWAYS make a new work container.

**Comment by milezhacks** [comment]
FWIW, I think Chrome nailed this part of the UX with their multi-user mode.  This binds a user to a window rather than a tab but the flow is always more natural.  
That said, one way to improve FF's container flow/experience (IMHO) is: if I hit alt/cmd-T to open a tab, it should spawn the same type of tab that I'm currently in.  

For me, I'm using containers to mix personal and work activities.  When I'm looking at a personal page and open a tab, chances are a want a personal container.  If I'm looking at a work tab and I spawn a tab, chances are I want a work container.

The last bit of behavior that breaks the flow is when I click a link from an external app.  If someone sends me a pull-request in Slack, I want that link to open in my work container.  How would I control that?  A link should open in the container type of the last viewed container.  So if I look at a work tab and click a link, I want a work container.  If someone sends me a l ink to reddit in iMessage, I'll look at a personal container and then click the link.

In the general case, all things boil down to :
```new container( type = typeof(lastviewed) )```

**Comment by Kylejustknows** [comment]
I want another kind of "Default Container" which I failed to get. Users can not define rules like wildcard.

I want google.com in Container1(for account reason), but everything else in Container2.
But now all the page I open from google.com(even I open in new tab) will keep using Container1, no way to switch containers automatically, no default container settings.

**Comment by eskemojoe007** [comment]
I actually love the one window, tabs of different types mix that containers allows, but I do wish that I could set the color and name of the "default" container.  AKA the default container shows up on the list, is customizable.

**Comment by RetroAsgardian** [comment]
An option to set the color of the "default" container would be nice, but an option to have links leave the container based on domain would be extremely useful, especially for isolating, say, Google, from the rest of my personal stuff. I don't use Facebook, and I don't trust that third-party "Google Container" addon.

**Comment by donghwicha** [comment]
Hi. Any update on this subject? I'm surprised to see many ppl are interested. 
Here is my use case 
- I want to use multi session container for gmail 
- gmail for company and gmail for my own 
- currently: everytime my new tab does not have any session container. so I have choose which session to use regardless of what I want 
- change: i set new tab default to my own gmail session. If I need I just change session container.

**Comment by andrewcrook** [comment]
The biggest issue that I have is not being able to set a default container (to act like a catch all) so that links to other sites  in a container don't open in the same container but rather to a default one, unless they are already configured to their own container that is. 

example:  clicking on a link in Twitter to a site that I have never visited before.

I agree on being able to see the default container in the list.

**Comment by rr0ss0rr** [comment]
I have "Temporary Containers" installed and like having my default container being a temporary one.  The container is removed when the last tab is closed.  I just wish this feature would be added to MAC

**Comment by donghwicha** [comment]
so how many upvote is needed?

**Comment by jonashaag** [comment]
@donghwicha Please think about how you communicate online. This is a free program; you are not paying any money, nobody owes you anything. Maybe you can try to implement it yourself and send a pull request?

**Comment by donghwicha** [comment]
> @donghwicha Please think about how you communicate online. This is a free program; you are not paying any money, nobody owes you anything. Maybe you can try to implement it yourself and send a pull request?

@jonashaag  I think it is you who should think about how you communicate online. This is a free program; we are not paying any money, nobody owes you anything, you don't own anything. Having communication is also part of participation, and asking how community works regarding valid number of vote is not offending anything. Please do not troll here throwing harsh expression like why don't you do it yourself?

**Comment by andrewcrook** [comment]
@donghwicha @jonashaag  please take this argument off this thread it doesn’t help anyone and isn’t what issues are for it just makes issues harder to manage. Also please take a look at the code of conduct for this project. Thank you.

https://www.mozilla.org/en-US/about/governance/policies/participation/

**Comment by DIW79** [comment]
I was wondering about this issue myself and had a look around Github.

I saw that there have been so very very many discussions and duplicates of this topic which made me recognise that there is _CLEARLY_ a need in the community for there to be some sort of option that allows people to select if they wish to open a new tab in a fully operating container!

Why this has been such a hard thing to at least get some practical movement on is a bit of a mystery to me, but seems, from the conversations I have read, to centre around what different people think the default container should be/how it should act. 

Unless I am barking up the wrong tree entirely (and I admit in advance that I may have misinterpreted things) it seems to me that this issue has been gone over time and time and time and time again ad infinitum to a point where the community seems to be at a standstill

I would like to propose what I think is a simple get-around (and which leaves open the potential for further development in terms of user-defined behaviour/functionality.

I propose the following:

1. That (simply to get things moving) an option is included to allow the user to specify whether they want new tabs to open in a container (one which functions as already existing containers with regards to security), or, whether they wish new tabs to open as normal (as if there were no container present). 

2. This user-defined functionality sets the behaviour of all new tabs by opening them in a 'new container tab' - maybe the new container tab has it's own colour not already listed - a rainbow line or black and white dashed line? 

3. This behaviour is set via a button which could be placed in the empty space next to the sort tabs button, and becomes highlighted when selected.

4. When this button is highlighted, all new tabs open in the 'new tab container', and when not highlighted, new tabs open as they would if not using multi-account containers. 

5. If the need then arises in the future (from further discussion) that different people wish such functionality to operate in differing ways then surely, the highlighted button could become a menu which opens to drop-down options to allow users to define the specific behaviour of the "new tab container".

**Comment by Solid-Ice8** [comment]
@DIW79, you can use a combination of add-ons to achieve what you're implying.

You should note that Multi-Account Container's "Default" is essentially no container at all. None. Not existing. Here's 2 add-ons I'll suggest to you and many users alike:

1) Temporary Containers; which I've explained here > #1722 
2) Simple Tabs Groups; which I've also explained here > #1695

**Comment by AlienResidents** [comment]
> I propose the following:
> .
> .
> .
>     7. Lastly, there will second and third items in the 'tab menu' accessed by left-clicking on a tab which offer the options to "reopen in default container" or simply "open new default container". I can see there also being the need to offer the option to "open new default container" in the container add-on menu (right click on multi-account containers icon in toolbar).

Thanks for your proposed ideas, @DIW79 .  I'd like to add an 8th option.

8. When using CLI tools, or anything external to the browser, that opens a page, I would like the option to select the container before loading the page requested from the external source. Perhaps in somewhere like the "new tab" page, and if FF is aware that an external source requested a page to be opened, then ask me first.  Maybe through a setting/toggle that gives me an option to say "open in the default container", "ask me" which container, or "select container from new tab page" etc.

A couple of real-world examples for me:
e.g. 1. I use CLI tools to authenticate myself for using a particular CLI tool (e.g. gcloud, gcs, awscli etc).  I have mutliple accounts, and so I'd like to select the container that I have tied to a particular account, instead of the default container.  I know it's easy enough to right click, and open in the correct container, but sometimes I have to catch the site before it fully loads in order to authenticate properly, which of course is a self-made race condition for how I use containers.
e.g. 2. I click a link in slack/discord/steam/uplay etc, and that opens in the default container (or the "remembered" container for that particular site/URL).

**Comment by Solid-Ice8** [comment]
> 2. I click a link in slack/discord/steam/uplay etc, and that opens in the default container (or the "remembered" container for that particular site/URL).

This is working as intended. You'll first need to assign a site to the container. Tick "Always open in X container".

Then, on a different container, navigate to that site. [It'll ask which container you want to open it in](https://i.postimg.cc/jdXnqfVV/Containers.png).




 Tick "Remember decision" (or something like that) and confirm the appropriate container.

**Comment by AlienResidents** [comment]
> > 2. I click a link in slack/discord/steam/uplay etc, and that opens in the default container (or the "remembered" container for that particular site/URL).
> 
> This is working as intended. You'll first need to assign a site to the container. Tick "Always open in X container".

This was an example usage based on the proposal I put forth, and while I understand you can "always open in X container", this isn't what I meant.  I'll try to explain this more in detail.

I use slack and discord , with different channels, and depending on the channel, I want an option to open github for instance, with the account based on the channel I click a link from.  I have my personal github account, and I have my work github account.  I can't always, open github.com in container X.

I would like an option where I don't have to re-open a tab in a different container, but instead be prompted for the container I'd like to open the site in, before it loads.

I hope this makes more sense.

**Comment by Solid-Ice8** [comment]
If that's the case, you can simply unassign GitHub from your current X container.

Locking a site to one specific container won't allow you to use multiple accounts.

This also applies to those add-ons like FaceBook & Google Containers. If you have any of those 2 installed in FireFox, you also won't be able to log into more than one account at a time.

The only other workaround is to make a new profile.

**Comment by AlienResidents** [comment]
> If that's the case, you can simply unassign GitHub from your current X container.
> 
> Locking a site to one specific container won't allow you to use multiple accounts.
> 
> This also applies to those add-ons like FaceBook & Google Containers. If you have any of those 2 installed in FireFox, you also won't be able to log into more than one account at a time.
> 
> The only other workaround is to make a new profile.

I'm fairly certain you've misunderstood me.  Maybe I have used bad examples.

In short, I'd like to be able to select a container to open an externally opened site/URL **before** the site/URL loads.  That behaviour, as far as I'm aware, does not exist, but I'd sure like it.

**Comment by Solid-Ice8** [comment]
> I'd like to be able to **_select a container_** to _open an externally opened site/URL before the site/URL loads._ That behaviour, as far as I'm aware, does not exist, but I'd sure like it.

I'm not sure if you're referring to history and/or bookmarks or something like using Google for example. You want to be able to "select X container" when Google redirects you to an external site correct?

**Comment by AlienResidents** [comment]
> > I'd like to be able to **_select a container_** to _open an externally opened site/URL before the site/URL loads._ That behaviour, as far as I'm aware, does not exist, but I'd sure like it.

ugh, no I'm referring to the initial comment I made, which I think you've forgotten about.

https://github.com/mozilla/multi-account-containers/issues/356#issuecomment-620942515

External to the browser, so the command line interface (CLI) which is external to the browser, Slack which is external to the browser, discord which is external to the browser.  The browser in this case is of course, Firefox.  External to the browser means, it didn't happen in the browser.  Something external to the browser makes the browser do something.  In this case, something external to the browser, tells Firefox to open a link, let's call that external something, in this example, Slack.  So again, Slack, which is external to the browser tells Firefox to open a link/site.  I want a choice to select which container Firefox uses, before Firefox opens the link/site that Slack requested.

At this point, I'm unsure how to make my request/proposal any more clear.  Happy to hear suggestions about how to approach this proposal differently.  If I had video, and animation skills I'd make a video.

**Comment by Solid-Ice8** [comment]
Thank you for clarifying and being more specific.

Since Slack is an external program, then it calls under:

>  Automatically reopens tabs in Temporary Containers when:

 - Opens a new tab
 - Tab tries to load a website in No Container
- **External program opens a link in the browser**

TC works with Slack. In advanced options, you can tick 'close leftover redirects" in Isolation. Slack is listed in there. You'll know more here.

Note that TC also has the ability to 'Convert Temp Container to Perm Container' (vice versa) & I've tested can confirm its compatible with MA-C.


I don't know if there's a real 'perfect solution' to _pre-select a container **before** FireFox launches_ but the closest thing I can think of is to make a container for those selections of external links from Slack. Tick "Always Open in Slack" container, but don't tick "Remember my decision for this site" yet (I'll explain below). Test & confirm it works with Slack.


If you tick "Remember my decision for this site", this will lock yourself out in case there's a need to use multiple accounts for any social media related sites.


And, also note:

I've had cases where MA-C is clashing with other containers add-ons like Google, Fb, TC & Containerise because "Always Open In" wasn't ticked & TC's temp container was over-lapping with MA-C's existing containers trying to assign it in a  new container, Thus crashing the tab in an infinite loop.

Basically, FireFox is saying:
MA-C has assigned this site to X container.
TC has also assigned "Temp" container.

Which one do I use? And thus confusing itself.

As soon as I ticked "Always Open In", the problem was resolved. I re-produced in TC / MA-C & solved it.

**Comment by Solid-Ice8** [comment]
I have read your post in #1727 and I'll leave my comments here as I don't want to post in multiple locations (and I'm sure that's also your intention)

I've been using Thunderbird, which sounds like a more simple version of Slack (based on my brief search on Google, it doesn't use emails like TB) but, this also means that external links clicked in emails will open anywhere as well, un-contained so to speak.

Let me know if the information I have is correct.

1) When Slack (external program, outside of FireFox) opens external links, you want to be able to select "Slack Team" or "Slack Workplace" containers.

2) Of course these will over-lap in websites, different identities (or logins) will be used.

3) In Slack, you click an external link in "Team", and FireFox should automatically open in "Team" container which has existing 'session' of logins and preferences.etc

4) Once again, in Slack, you click an external links in "Workplace", FireFox starts and "Workplace" container opens automatically with different existing 'session'

I've made a diagram to show my understandings. I'm no artist or anything, so don't mind me.

https://i.postimg.cc/pTWRLyzr/Slack-Fire-Fox.png

Feel free to edit anything that's incorrect.

If CLI = command line interface then, you'd need something that will intercept between Slack and FireFox. 

You'll notice I put text along the linking area to Slack & FireFox boxes.

In my opinion, this goes beyond just FireFox (MA-C) and Slack. This is what a power user would want.

Sure, you can make containers matching multiple account logins with MA-C, but you'll need:

1) Something that will specifically say "These belong in 'Team' container" and "These belong in "Workplace" container.

2) If these belong in "Team" container, then launch FireFox and start "Team" container. Same with Workplace container.

3) The real question is: How will FireFox automatically start in these specific containers?  Whenever you close FireFox, the whole session is saved including other containers. How will FireFox "know" which containers are opened and which container is currently active?

Even with the add-ons I'm familiar with, they won't even come close to doing that. Like I said, I use TB and I've played around all day trying to figure it out. 

I noticed if you bookmark the main site and have ticked "Always Open In", but not having "Remember my decision" ticked will give you more of a choice to choose which container you want to open. Still that's manual and not automatic.


The only workaround I can think of is to create another FireFox profile. Name them accordingly. When Slack clicks a link in "Team", it'll open up FireFox "Team" Profile. And when clicking links in Workplace, it'll open in Workplace Profile.

-----------------------------------------------------------------

I will now talk in code language....if I can:

Slack : Clicked external link in Team > Do these links belong to Team or Workplace? 

If Team, then execute: FireFox in Team Profile
If Workplace, execute: FireFox in Workplace Profile.

**Comment by andrewcrook** [comment]
Guys this is really going off topic from the opening post perhaps remove and raise a new ticket?

**Comment by Solid-Ice8** [comment]
@andrewcrook, Okay, I'll post in his new topic (#1727) if there's anything else.

**Comment by robmonte** [comment]
It's pretty appalling that 3 years have passed with no progress on this. I searched around a bit before requesting this feature myself only to find it's been requested many times beforehand already.

I am sick of opening new tabs in a container I have set for another website, particularly Google. I have a Google container, and if I am googling something and click a search result, I do not want to visit that site in my Google container. To me, this completely defeats the purpose of having Google in said container. We need a setting akin to "open all non-containered websites in a 'default' container" in this addon.

**Comment by DIW79** [comment]
> It's pretty appalling that 3 years have passed with no progress on this. I searched around a bit before requesting this feature myself only to find it's been requested many times beforehand already.
> 
> I am sick of opening new tabs in a container I have set for another website, particularly Google. I have a Google container, and if I am googling something and click a search result, I do not want to visit that site in my Google container. To me, this completely defeats the purpose of having Google in said container. We need a setting akin to "open all non-containered websites in a 'default' container" in this addon.
> 
> Does Mozilla even bother looking at these posts?

I wholeheartedly agree with everything @robmonte says here. I've even asked on other threads and been told 'use a different add on', but a) that beats the point of having one brilliant add on that does everything and b) the suggested add-on doesn't actually work in the intended fashion and is only a clunky stop-gap at best.

**Comment by andrewcrook** [comment]
@DIW79 @robmonte 

I don't think this is an official product it started as  a research experiment apart of Test Pilot. So it probably doesn’t get the same resources as the official products. 

However, it is community driven and open source so, if you can program, contribute and start on some of these issues. It would be far more helpful than moaning.

**Comment by Solid-Ice8** [comment]
@DIW79 @robmonte 

Nothing in the world is perfect.

I guess I'm the one to blame for being told to "use a different add-on". However:

1) "use a different add-on" is only the workaround solution after X years (which some of you have noticed) project has been abandoned. 

2) > I am sick of opening new tabs in a container I have set for another website, particularly Google. I have a Google container, and if I am googling something and click a search result, I do not want to visit that site in my Google container. To me, this completely defeats the purpose of having Google in said container. We need a setting akin to "open all non-containered websites in a 'default' container" in this add-on.

You should, by now understand that Multi-Account Container's "Default container" literally means uncontained tab, _there is literally no container_. It does not mean it's a 'No Container' container. No container is present at all. **This is the main reason the sites are opening in Google container**.

This is the reason I suggest other add-ons; like TC. I've tested the quoted text above isn't the behaviour for me. Whenever I visit a site outside Google, this will open a new temporary container tab. Naturally, this also means I've installed the actual add-on "Google Container" as well.

I've taken "google.com" out of my MA-C's Google container for this to work.

If you would like a video proof, I'll be more than happy to make one as confirmation.

And I also understand a 'Default Container' should be added, like the default download directory is always "Downloads" folder for FireFox (of course, unless you change it).

I'm taking steps to re-learn programming due to quarantine life-style while my occupation is outside programming.

**Comment by robmonte** [comment]
> You should, by now understand that Multi-Account Container's "Default container" literally means uncontained tab, _there is literally no container_. It does not mean it's a 'No Container' container. No container is present at all. **This is the main reason the sites are opening in Google container**.

"No container" and a "default" container would be indistinguishable in usage from the user's POV. The end result would be the same either way. I called it a 'no container' container for simplicity's sake.

**Comment by kineita** [comment]
I like this feature. Example: I browsing web with Google container, but I click a hyperlink which is not in Google container so it will open "Default" or "Temp" container.
Hope it come soon.

**Comment by donghwicha** [comment]
It is pretty sad to see the way how Firefox is managed. The kind of massage like "If you want it, go and f**k it by yourself" makes people disappointed. It would better to just be honest that 
- firefox team don't have enough resource to manage this plugin in
- firefox team believe that this can be seed of another plugin with more possibilities  

It is just sad for you guys to puke on people. If you all can not maintain this repository, just shut it off.

**Comment by terrycloth** [comment]
It would be hard for a potential contributor to develop a pull request that would satisfy this issue, because it's not very clear what this issue is even about. It doesn't help that a lot of the comments have just been angrily impatient, but also a lot of the "+1"-type comments are actually asking for something different. It might make sense to split this into separate issues for discussing the different specific requests.

**Comment by Solid-Ice8** [comment]
I (& a few others) have continuously suggested "Simple Tab Groups" for users asking for a type of "default container" and this add-on will also solve #1749.

I'm sorry to say and I hate to be rude/harsh but most users don't actually "explore" add-ons.

In this add-on, users can create a new "group" and then set it to "Main Group" at the bottom of 'Group Settings'. This will force all external links opening into that main group.

They can "catch" all of one container into this group or even multiple containers to auto-move into the group. This is a more advanced version of MAC's Sort Tabs.

They can also individually 'Sticky Group', which basically means locking the group. Tabs will not 'catch and move' to other groups if the same container(s) are in the group.

And finally, they can even use expressions for wildcards (*) domains.

I've basically suggested pretty much an all-in-one add-on for them.

**Comment by andrewcrook** [comment]
@Solid-Ice8  STG still doesn’t seem to open duckduckgo links into a default container I created and keep all tabs visible. I can make it move those links to a different group via domain matching giving them the default container. But I have to keep switching groups to view the tabs I would like all the tabs displayed at once keeping those groups otherwise it becomes annoying.

**Comment by Solid-Ice8** [comment]
I use DDG as well. I also use Temp Containers, you're correct. It doesn't open a new container tab like it does with Google search engine.

This add-on is a bit tricky as it has "isolation" methods which you'd have to do a one-time manual set-up of adding the pop-up version which appears at the end of URL to disable TC to add new domains into MAC. And you'll have the option of using keyboard shortcuts and a few other things in "Options" and there's "advanced options" as well.

You can enable "Alt+C" - which opens new Temp Container and "Alt+I (Capitalized i) and that's for disabling TC isolation.  There's a few other shortcuts I haven't mentioned, so you can have a look.

And, it can also convert temp container tabs into perm container tabs, also from the pop-up; under "Actions" or the 3 lined hamburger menu (as Mozilla FireFox calls it).

**Comment by groovecoder** [comment]
Note: while 7.0.0 does not include this feature per sé, it does include "Site Isolation" which can help with some of the use-cases for this feature. E.g., isolating google.com into a container, so that the result links are opened into different container tabs.

**Comment by grahamperrin** [comment]
> … set a default container for new tabs …

Now, will a single default container be substantially more desirable than a single default _non_-container (i.e a normal tab)? 

Consider the features of recent releases of Firefox: 

* [Firefox 85 Cracks Down on Supercookies - Mozilla Security Blog](https://blog.mozilla.org/security/2021/01/26/supercookie-protections/)
* [Firefox 86 Introduces Total Cookie Protection - Mozilla Security Blog](https://blog.mozilla.org/security/2021/02/23/total-cookie-protection/)
* [Introducing State Partitioning - Mozilla Hacks - the Web developer blog](https://hacks.mozilla.org/2021/02/introducing-state-partitioning/)

----

Food for thought: [Always In Container](https://addons.mozilla.org/addon/always-in-container/) can ensure use of a container …

**Comment by grahamperrin** [comment]
From highest-rated comment <https://github.com/mozilla/multi-account-containers/issues/356#issuecomment-358723450> 

> … the big thing I'm hoping for is ctrl-t ALWAYS make a new work container. …

It appears that you can have one shortcut per container, up to ten: 

![image](https://user-images.githubusercontent.com/192271/115940728-89474400-a49a-11eb-8461-3d937e927f01.png)

**Comment by br4nnigan** [comment]
Excuse me for not reading the whole thread, but it seems that people want a choice to open new tabs in a container.

Personally I want the opposite: what annoys me is that a new tab (ctrl+t) is always opened in the container (or non container) that the current tab is in.

I should be able to set new containers to always open in a non-container. 

@grahamperrin Hey I know you from the waterfox reddit! 👋 It's the browser I actually use

**Comment by remram44** [comment]
This is still a problem. "No container" is a de-facto container: you can have tabs in it, it stores cookies and session, etc. Yet it is present as an option in some contexts (e.g. "open in new container tab" shows "no container") but is painfully missing in others (e.g. no keyboard shortcut to open in "no container" container, not present in "always open this site in container").

I don't care whether the "no container" can be replaced by a "default container" (which would be available in all those contexts) or whether the "no container" container should just be added to those places as the container that it is. Either way would fix those many pain points for me. But the current situation is unreliable, inconsistent, and frustrating.

**Comment by synth** [comment]
I came back to Firefox after a decade or more (as others have stated) just for this containers feature. So, its truly a bummer to see that it doesn't really work as I would expect - which is that containers should be "sticky". 

Imagine you have two browser windows open, one is focused. You switch to a desktop app and click a link. The new tab will open in the browser window that most recently had focus. This is dependable and expected behavior. Containers should follow suit. 

If I cmd/ctrl-t, I would expect a new tab in the same context of the most recently (or currently) focused tab. If I click a link from a desktop app, the same thing should happen, a new tab should open in the most recently focused browser window and retain the container context of the most recently focused tab. To me, this is a better implementation than a named container that functions as a default as it follows existing browser/OS patterns. 

Without this, this feature seems unusable for me as I would have cognitive load trying to ensure that I'm always in the correct container as I cmd-t dozens of times a day causing me to leave my expected container context or having to move tabs across which is unnecessary work. 

I would happily submit a PR if I knew where to start or sponsor a developer to build this. This could even be a setting: "Sticky Containers".

**Comment by br4nnigan** [comment]
> If I cmd/ctrl-t, I would expect a new tab in the same context of the most recently (or currently) focused tab

No, why would you expect that the new url you want to load is related to the currently focused tab? This would be highly annoying as usually you'd want a new tab to use no/the default container. I'm already having this exact problem with an addon that I use that uses the current container for new tab and it's highly annoying (TabSidebar, open tab by double click). _Thankfully_ I can ctrl+t to open a tab without container.

> Without this, this feature seems unusable for me as I would have cognitive load trying to ensure that I'm always in the correct container as I cmd-t dozens of times a day causing me to leave my expected container context or having to move tabs across which is unnecessary work.

No, **with** this, I would have cognitive load trying to ensure that the last container I was in (20 minutes ago?) suits some random link that I click. I would expect that link to open in the default container.

If you want certain urls opened within specific containers you can already ensure that by binding a certain container to a certain url.

Otherwise links should open in the default container imo.

**Comment by synth** [comment]
Exactly. I respect that you want it one way. I want it another. It should be a setting.

**Comment by remram44** [comment]
> If I cmd/ctrl-t, I would expect a new tab in the same context of the most recently (or currently) focused tab.

This is the behavior I'm experiencing. In any case you can get this by pressing a different key combination (e.g. Ctrl+Shift+1 instead of Ctrl+T). My problem is that there is no key combination for opening a tab in the default container (maybe Ctrl+Shift+0 should do this?)

**Comment by adrianmace** [comment]
To throw in my 2c.

I also have just come back to Firefox after a decade or more of Chromium. I am looking to replicate what Chrome does with their Profiles feature.

I want one window full of 'personal' tabs and one window full of 'work' tabs with almost no way to accidentally mix the two up.

I guess I need to change my workflow so that "default" on my work device is my 'work' container, and "default" on my personal device is my 'personal' container, but this means I cannot sync the settings between the two.

**Comment by remram44** [comment]
Firefox has profiles, go to `about:profiles` and use the "launch profile in new browser" button. https://support.mozilla.org/en-US/kb/profile-manager-create-remove-switch-firefox-profiles

**Comment by achernyakevich-sc** [comment]
> I came back to Firefox after a decade or more (as others have stated) just for this containers feature. So, its truly a bummer to see that it doesn't really work as I would expect - which is that containers should be "sticky".
> 
> Imagine you have two browser windows open, one is focused. You switch to a desktop app and click a link. The new tab will open in the browser window that most recently had focus. This is dependable and expected behavior. Containers should follow suit.

@synth I would propose a solution that could behave similar as you described. It will be based on usage of Simple Tab Groups (STG) add-on together with Multi-Account Containers (MAC) add-on that could be greatly integrated. To achieve having two windows every with own container that by default will open tab in the dedicated container you need to do something like:

* Created in MAC containers C-A and C-B.
* Create in STG a group of tabs G-A and configured it as "sticky" so that this group's new tabs always will be open in the container C-A.
* Create in STG a group of tabs G-B and configured it as "sticky" so that this group's new tabs always will be open in the container C-B.
* Open new window in Firefox, switch it to show the G-A group of tabs.
* Open new window in Firefox, switch it to show the G-B group of tabs.

As result you will have two windows and every of them will be sticky to their container.

To simplify you life I would propose never close these windows, instead you could use Ctrl+Shift+Q (or Cmd+Q) to close all windows of Firefox. So when Firefox will be started again you will have have restored all earlier closed windows. Alternative could be loading any other group to window that you plan to close. Otherwise you potentially could lose tabs earlier open in the group.

**Comment by synth** [comment]
@achernyakevich-sc Interesting! Thanks for the workaround! I'll check it out. 🤝
